packages/alfa-rules/src/sia-r82/rule.ts

```diff
@@ -0,0 +1,188 @@
+import { Rule, Interview } from "@siteimprove/alfa-act";
+import { Element, Namespace, Node } from "@siteimprove/alfa-dom";
+import { Predicate } from "@siteimprove/alfa-predicate";
+import { Option } from "@siteimprove/alfa-option";
+import { Result, Ok, Err } from "@siteimprove/alfa-result";
+import { Page } from "@siteimprove/alfa-web";
+
+import { expectation } from "../common/expectation";
+
+import { hasRole } from "../common/predicate/has-role";
+import { isPerceivable } from "../common/predicate/is-perceivable";
+
+import { Question } from "../common/question";
+
+const { isElement, hasNamespace } = Element;
+const { and, test } = Predicate;
+
+export default Rule.Atomic.of<Page, Element, Question>({
+  uri: "https://siteimprove.github.io/sanshikan/rules/sia-r82.html",
+  evaluate({ device, document }) {
+    return {
+      applicability() {
+        return document
+          .descendants({ flattened: true, nested: true })
+          .filter(
+            and(
+              isElement,
+              and(
+                hasNamespace(Namespace.HTML),
+                hasRole(
+                  "checkbox",
+                  "combobox",
+                  "listbox",
+                  "menuitemcheckbox",
+                  "menuitemradio",
+                  "radio",
+                  "searchbox",
+                  "slider",
+                  "spinbutton",
+                  "switch",
+                  "textbox"
```

Comment:
I'm not entirely convinced as it currently aligns well with the actual rule definition. This will hopefully change once we fork the rule in Sanshikan and come up with a good term for "semantic role that inherits from the abstract input or select role, and
does not have a required context role that inherits from the abstract input or select role".